### PR TITLE
Receive submissions send results

### PR DESCRIPTION
**Pull request by amohamed97**
now accepts submissions file, thoroughly validates it, then proceeds to run the grader with the preexisting submissions, because extraction isn't yet implemented
after grading the lab a result file with the name of the lab is added to "res" directory, and the results can be fetched later through /results/<course_id>/<lab_id>:GET endpoint

### CODE REVIEW

**Comment by amohamed97** [comment]
@shakram02 the conflicts have been resolved

**Comment by amohamed97** [comment]
@Hagar-Usama you can continue from here and make a pull request with the extraction support

File: .gitignore [hunk 1]
```diff
@@ -138,4 +138,11 @@ dmypy.json
...
-.vscode
\ No newline at end of file
+.vscode
+
+#Courses
+**/cc451/*
+
+#results
+output.txt
+parser_output
```
Comment (shakram02) [inline .gitignore:148]: Please make a PR with this gitignore file. 

https://github.com/github/gitignore/blob/master/Python.gitignore

**Review by shakram02 — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (amohamed97) [reply .gitignore:148]: I checked the diff between the .gitignore in this branch and the one in the provided link and they are identical, except for a few extra lines in the one in this branch.

**Comment by shakram02** [comment]
please resolve

**Review by Hagar-Usama — APPROVED** [review]

**Review by shakram02 — APPROVED** [review]

**Pull request merged**